### DISCUSSION TITLE
Destroy vehicle API call and pawn path

### DISCUSSION THREAD

**Discussion by Sabrin0** [microsoft/AirSim#4227] [Support Q&A]
HI, the python method `simAddVehicle` allow to spawn a vehicle in the simulation during runtime. Is there also a method to destroy the vehicle? Moreover the aforementioned method takes as argument the pawn path, I tried with `'/AirSim/Blueprints/BP_DummyDrone.BP_DummyDrone_C'` and similar but I got an exception:

Unhandled exception at 0x00007FFE5D864F69 in UE4Editor-Win64-DebugGame.exe: Microsoft C++ exception: std::out_of_range at memory location 0x000000D46CD7A730.

**Comment by zimmy87** [comment]
It's important to note that the `pawn_path` parameter of `simAddVehicle` doesn't refer to the literal reference path of the custom blueprint. Instead, you need to define a new PawnPath in the settings.json (see https://microsoft.github.io/AirSim/settings/#pawnpaths) that is referencing your custom blueprint, then pass the name of your new PawnPath in the settings to your `simAddVehicle` call.

As for destroying the vehicle, we currently do not have an API to handle that operation. I tried using `simDestroyObject` locally, but that resulted in a crash when I passed the name of a drone. This would be a good idea for a new PR though.

**Reply by Sabrin0** [reply]
Thank you for your answer @zimmy87. If necessary I'll try to implement a new api call
